tests/test_cli_compile.py

```diff
@@ -90,13 +87,42 @@ def test_command_line_setuptools_read(pip_conf, cli_args):
             setup(install_requires=[])
         """))
         package.close()
-        out = runner.invoke(cli, cli_args)
+        out = runner.invoke(cli)
 
         # check that pip-compile generated a configuration
         assert 'This file is autogenerated by pip-compile' in out.output
         assert os.path.exists('requirements.txt')
 
 
+@pytest.mark.usefixtures('pip_conf')
+@pytest.mark.parametrize('options, expected_output_file', [
+    # For the `pip-compile` output file should be "requirements.txt"
+    ([], 'requirements.txt'),
+    # For the `pip-compile --output-file=output.txt` output file should be "output.txt"
+    (['--output-file', 'output.txt'], 'output.txt'),
+    # For the `pip-compile setup.py` output file should be "requirements.txt"
+    (['setup.py'], 'requirements.txt'),
+    # For the `pip-compile setup.py --output-file=output.txt` output file should be "output.txt"
+    (['setup.py', '--output-file', 'output.txt'], 'output.txt'),
+])
+def test_command_line_setuptools_output_file(options, expected_output_file):
+    """
+    Test the output files for setup.py as a requirement file.
+    """
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        package = open('setup.py', 'w')
+        package.write(dedent("""\
+            from setuptools import setup
+            setup(install_requires=[])
+        """))
+        package.close()
+
+        out = runner.invoke(cli, options)
+        assert out.exit_code == 0
+        assert os.path.exists(expected_output_file)
```

Comment:
Just a nitpick: the asserts could be dedented.

Reply:
Agreed. Let's clean it up in a following PR.

Reply:
In addition to that, i would also make the runner a fixture and clean up all the tests (which use the runnrt) like this one: https://github.com/jazzband/pip-tools/blob/67edc7d227507f3902090d2db8e129715279ae04/tests/test_cli_compile.py#L442-L449

IMO the tests look more concise in that way.